### ISSUE TITLE
Review use of `humanize` in templates

### ISSUE THREAD

**Issue by nlhkabu** [pypa/warehouse#6312]
From #6079
> For dates in the "Release date" column, use Jinja2's localize instead of humanize to insert proper ARIA tags

We are using the `humanize` function in several places.  We should review this on all templates in line with this recommendation

**Comment by di** [comment]
These are both macros that we've defined:

https://github.com/pypa/warehouse/blob/4229368158472164ddebc221acf101bbbf75019b/warehouse/templates/base.html#L15-L25

We should be able to add the necessary tags to the `time` element here instead, so we still retain the ability to "humanize" these timestamps.

**Comment by nlhkabu** [comment]
Note that when using 'humanize', we should also expose the full datetime in a tooltip (available on hover)

**Comment by nlhkabu** [comment]
@woodruffw what aria attributes were you recommending in your original observation?

**Comment by woodruffw** [comment]
@nlhkabu I was thinking `aria-label` originally, but I've realized that doesn't make sense in this context. The other lingering issue with `humanize` instead of `localize` is that it doesn't emit semantic `<time>` elements (and doesn't include tooltips, like you already mentioned).

**Comment by nlhkabu** [comment]
@bmustill-rose this is something I forgot to test with you yesterday :(

Could you tell me - what is the experience using a screenreader to read dates on PyPI?  Do you have any recommendations on how we could improve this (or websites that do this well)?

Thanks

**Comment by bmustill-rose** [comment]
Apologies for the delay.

They seem fine (as in they read as I'd expect them to having looked at the code) although you shouldn't need to set an aria-label if the information you want spoken is the same information that's surfaced to sighted users. Aria-label is mainly useful in scenarios where a text label isn't available or for whatever reason you want the screen reader to speak something different to what's being displayed.
Having said this, it's not always possible to access tooltips using a screen reader, so there's probably an argument to say that the aria label should be set to the full date & time so that there's a way for screen reader users to access that information.

**Comment by nlhkabu** [comment]
Thanks @bmustill-rose.  Note: we should complete https://github.com/pypa/warehouse/issues/6480 before this ticket, to ensure our tooltips are accessible.

**Comment by nlhkabu** [comment]
I've been looking over this and believe this is the best strategy to move forward:

- Make all dates use the same macro
  - Allow devs to disable/enable 'relative time' (e.g. "2 hours ago") via the macro parameters
  - Allow devs to show/hide the time (again, via the macro parameters)
- As per the current behaviour, only make dates relative when within the limit of the current day (e.g. "today", "2 hours ago"). 
- Add `iso 8601` datetime as a `title` (available to desktop users on hover)
- Add `iso 8601` datetime as the `aria-label` (so that screen reader users get all of the information)

Re semantics: the outputted HTML should meet the HTML5 specification by emitting a `<time>` element with a valid `datetime` value. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time for examples.

Based on our current codebase, I believe that in most cases we should display just the date by default, with relative time enabled.  The only times when this might not be appropriate are on the security history and journal pages, where users probably want to see the full date and time, without having to hover to get this information from the date title.
